connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -151,6 +151,8 @@ public void removeMetrics() {
         taskMetricsGroup.close();
     }
 
+    protected abstract void initializeAndStart();
```

Comment:
I see how you've just pulled the `WorkerSinkTask.initializeAndStart()` method up to this abstract class (per @C0urante's suggestion. That is a nice clean way to encapsulate that logic into the base class and keep the metrics behavior hidden from the subclasses. Nicely done.

It is a tiny bit unfortunate that the tests need to do something like:
```
        workerSourceTask.initialize(TASK_CONFIG);   // This just sets the config on the worker task
        workerSourceTask.initializeAndStart();            // This calls task.initialize(...) and task.start(...)
        workerSourceTask.execute();
```
But the `initializeAndStart()` method in the `WorkerSinkTask` has been around since the beginning, and it's probably not worth changing here. After all, using `initializeAndStart()` still makes sense _within_ the `WorkerSinkTask` and now also the `WorkerTask` and `WorkerSourceTask` classes. So I'm fine with enlisting that method name as-is.